src/ros_license_toolkit/checks.py

```diff
@@ -337,3 +337,28 @@ def _print_info(self, info_str, files_with_uncovered_licenses,
                     lambda x: x[0] in files_with_uncovered_licenses or (
                         x[0] in files_not_matched_by_any_license_tag),
                     package.found_files_w_licenses.items()))))
+
+
+class LicenseFilesReferencedCheck(Check):
+    """Check if all found License file have a reference in package.xml."""
+
+    def _check(self, package: Package):
+        not_covered_texts: Dict[str, str] = {}
+        for filename, license_text in package.found_license_texts.items():
+            # skipping all declarations above the package
+            if '../' in filename:
+                continue
+            if license_text['detected_license_expression_spdx'] not in \
+               package.license_tags:
```

Comment:
What would happen if the tag would be a non-spdx tag?

Reply:
package.found_license_texts should contain just the scan results that are spdx conform.
I'll add a check for 'detected_license_expression_spdx' nevertheless.